### PR TITLE
Added three modules to calculate the surface charge density of a sphere of radius (R) held at constant potential (V) and test for it.

### PR DESCRIPTION
**Pull request by HeshamElsaman**
# Pull Request Templating for Beginners #
The most common way **YOU** will interact with, and submit material to any repository you work on is via a "Pull Request".
Now not just any old Pull Request will do however; because hard at work making sure your repository stays in tip-top shape are the wonderful maintainers and reviewers whose task it is to review pull requests and maintain the repository so that they meet up-to-date regulatory standards.
If you follow this simple guide you too can be submitting Grade A pull requests that will have reviewers saying "Gee this pull request sure is swell. **Approved!**"

To ensure that your pull request is a real winner make sure it includes the following:
  - A legible title that describes what is contained within the pull request.
  - A legible description that goes into more detail than your title regarding what each file contained within the request is and does.
  - A reference to any currently open issues that pertain to your pull request.
  - And make sure to label and add a milestone to your pull request.

If you have followed these steps you're on your way to a successful pull request.
However, today's youthful programmers also need to format the file structure of their pull requests so the repository can remain clean and organized.

The general structure of your pull request should be the following:
  - A directory folder labeled in accordance with your assigned task.
    - Within should contain:
    - Two or more separate module files containing your function algorithms titled after each, and the appropriate docstrings.
    - One or more files containing the unit tests for your algorithms and the appropriate unit test title.
    - Finally, for clarity, you can add a helpful README.md that further describes your algorithms, their tests, and how to run them.
   
With this, you now have the makings of a solid and respectable pull request.

So get out there champ and start submitting **TODAY**.

*Any distress or bodily harm that may come as a result of the reviewing process or by the maintainer is not the responsibility of said reviewers and maintainers.*

So the first task for a successful pull request is: 

**remove now this template text and write a good description of what your pull request is about in its place**

### CODE REVIEW

**Comment by laserlab** [comment]
@iglesias-cardinale Please review asap

**Review by iglesias-cardinale — COMMENTED** [review]
Both the physics function and the mathematical operations are in their own modules. 

Every function includes a docstring with the input parameters and return. 

UnitTestingModule.py tests good cases of V=0 and $R\rightarrow\infty$ and all tests are passed when run

UnitTestingModule.py uses the assert statement

PhysicsModule.py raises a syntax warning when run in Terminal because of the use of the \e in \epsilon_0, which is interpreted as an invalid escape sequence. This, however, isn't a critical error since it's a syntax warning in a comment and the code still runs as intended.

PhysicsModule.py uses a lambda function and UnitTestingModule.py uses list comprehension, satisfying the homework requirement.

The PR Request statement is a template for a PR Request. This should probably be changed to be more representative of the physics and code involved.

File: SphereSurfaceChargeDensity/MathOpsModule.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def div(numerator, denominator):
+    """
+    Returns the result of dividing two numbers
+    Parameters:
+    numerator (float), denominator (float): input numbers
+    Returns:
+    number (float): The result of dividing the first number by the second number utilizing the lambda expression
+    """
+    Div = lambda x, y: x/y
```
Comment (laserlab) [inline SphereSurfaceChargeDensity/MathOpsModule.py:11]: Do not use your function name and a variable name with different cases. See linting

File: SphereSurfaceChargeDensity/MathOpsModule.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+def mult(x, y):
+    """
+    Returns the result of dividing two numbers
+    Parameters:
+    x (float), y (float): input numbers
+    Returns:
+    number (float): The result of multiplying the two numbers utilizing the lambda expression
+    """
+    Mult = lambda xx, yy: xx*yy
```
Comment (laserlab) [inline SphereSurfaceChargeDensity/MathOpsModule.py:22]: Do not use your function name and a variable name with different cases. See linting

File: SphereSurfaceChargeDensity/MathOpsModule.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+""" This Module is to define the general mathematical operations to be used by the other modules """
+
+def div(numerator, denominator):
+    """
+    Returns the result of dividing two numbers
+    Parameters:
+    numerator (float), denominator (float): input numbers
```
Comment (laserlab) [inline SphereSurfaceChargeDensity/MathOpsModule.py:7]: if float is not actually required or checked do not use it in the docstring

File: SphereSurfaceChargeDensity/PhysicsModule.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    The constant potential (V): a float number.
+
+Considering the units used, the default system will be the metric system.
+However, it will be flexible to convert to other systems by manually choosing the inpus' units.
+A constant value of \epsilon_0 is thus set by default to the units of the metric system.
+"""
+
+from MathOpsModule import *
+
+def sigma(R, V, epsilon_0 = 8.854*(10**(-12))):
```
Comment (laserlab) [inline SphereSurfaceChargeDensity/PhysicsModule.py:16]: CONSTANTS should not be function arguments

File: SphereSurfaceChargeDensity/PhysicsModule.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    Returns the surface charge density of a conducting spherical shell of radius (R) held at constant potential (V).
+    Parameters:
+    R (float), V (float), epsilon_0 (float): input numbers (radius, potential, electric permitivity of free space)
+    Returns:
+    number (float): the surface charge density
+    """
+    Sigma = lambda epsilon_00, V0, R0: mult(epsilon_00, div(V0, R0))
```
Comment (laserlab) [inline SphereSurfaceChargeDensity/PhysicsModule.py:24]: I very much agree with the linter here that this is not a use case for a lambda function

**Review by laserlab — CHANGES_REQUESTED** [review]
While Sigma is used inside sigma this is definitely not a useful case for using a function inside a function.

File: SphereSurfaceChargeDensity/PhysicsModule.py (hunk 4 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/PhysicsModule.py:16]: That was intended to be flexible with the choice of the systems of units. So, programming-wise, it's just a variable with a default value.

File: SphereSurfaceChargeDensity/MathOpsModule.py (hunk 3 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/MathOpsModule.py:7]: Noted.

File: SphereSurfaceChargeDensity/MathOpsModule.py (hunk 1 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/MathOpsModule.py:11]: I don't quite get it.

File: SphereSurfaceChargeDensity/PhysicsModule.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+This Module is to define the main desired funtion:
+    A function to calculate the surface charge density of a conducting spherical shell held at constant potential.
+
+The specifications needed for the shell:
+    The radius (R): a float number.
+    The constant potential (V): a float number.
+
+Considering the units used, the default system will be the metric system.
+However, it will be flexible to convert to other systems by manually choosing the inpus' units.
+A constant value of \epsilon_0 is thus set by default to the units of the metric system.
```
[the remark this answers is not in the document]
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/PhysicsModule.py]: Removed.

File: SphereSurfaceChargeDensity/MathOpsModule.py (hunk 1 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/MathOpsModule.py]: Removed.

File: SphereSurfaceChargeDensity/MathOpsModule.py (hunk 2 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/MathOpsModule.py]: Removed.

File: SphereSurfaceChargeDensity/PhysicsModule.py (hunk 5 above)
Comment (HeshamElsaman) [reply SphereSurfaceChargeDensity/PhysicsModule.py]: Removed.

**Review by laserlab — APPROVED** [review]

**Comment by laserlab** [comment]
merging

File: SphereSurfaceChargeDensity/PhysicsModule.py (hunk 4 above)
Comment (laserlab) [reply SphereSurfaceChargeDensity/PhysicsModule.py]: Since there is a limited amount of possibilities a list of CONSTANTS would be suitable

Comment (laserlab) [reply SphereSurfaceChargeDensity/PhysicsModule.py]: You also probably do not want to change that during runtime which makes this a CONSTANT in any case

**Pull request merged**